ui/components/app/name/name-details/name-details.tsx

```diff
@@ -86,6 +79,24 @@ function formatValue(value: string, type: NameType): string {
   }
 }
 
+// Provider source ids that have a localized name:
+const LOCALIZED_PROVIDERS = ['ens', 'etherscan', 'lens', 'token'];
+// The following lines avoid i18n "unused messages" lint error.
+// t('nameProvider_ens');
+// t('nameProvider_etherscan');
+// t('nameProvider_lens');
+// t('nameProvider_token');
+
+function getProviderLabel(
+  sourceId: string,
+  t: ReturnType<typeof useI18nContext>,
+) {
+  if (LOCALIZED_PROVIDERS.includes(sourceId)) {
+    return t(`nameProvider_${sourceId}`);
+  }
+  return sourceId;
```

Comment:
I think we still need to also check the original `nameSources` state.

In the case of snaps, this will currently result in `npm:someSnap` for example rather than a friendly name like `Some Awesome Name Resolution Snap`.

Reply:
Done. I added test coverage for this fallback and also the "Recognized name" state.